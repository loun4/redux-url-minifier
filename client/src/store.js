
import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { createForms } from 'react-redux-form';
import entities from './reducers/entities';

const loggerMiddleware = createLogger({
  predicate() { return process.env.NODE_ENV === 'development'; },
});

const forms = createForms({
  linkForm: {
    linkURL: '',
  },
  signinForm: {
    login: '',
    password: '',
  },
});

export default function configureStore(preloadedState) {
  return createStore(
    combineReducers({
      ...forms,
      entities,
    }),
    preloadedState,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware,
    ),
  );
}
