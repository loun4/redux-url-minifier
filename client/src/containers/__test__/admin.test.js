
import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RouterProvider from '../router-provider';
import store from '../../store';
import ConnectedAdmin, { Admin } from '../admin';

let subject;
const testStore = store();

const buildSubject = props => (
  <RouterProvider>
    <Provider store={testStore}>
      <ConnectedAdmin {...props} />
    </Provider>
  </RouterProvider>
);

beforeEach(() => {
  subject = buildSubject();
});

test('Render admin', () => {
  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render loading icon on fetching', () => {
  const s = buildSubject({
    session: {
      ...testStore.getState().session,
      isFetching: true,
    },
  });

  const component = renderer.create(s);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render loading icon if session is ready for auth', () => {
  const s = buildSubject({
    session: {
      ...testStore.getState().session,
      readyToAuthenticate: true,
    },
  });

  const component = renderer.create(s);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render signin form is deauthenticated', () => {
  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render link form & table if authenticated', () => {
  const s = buildSubject({
    session: {
      ...testStore.getState().session,
      isAuthenticated: true,
    },
  });

  const component = renderer.create(s);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render call fetchEntityData', () => {
  const props = mount(subject).find(Admin).first().props();
  const fetchEntityData = jest.fn();
  const component = shallow(<Admin {...props} fetchEntityData={fetchEntityData} />);

  component.setProps({
    session: {
      ...testStore.getState().session,
      isAuthenticated: true,
    },
  });

  expect(fetchEntityData).toHaveBeenCalled();
});
