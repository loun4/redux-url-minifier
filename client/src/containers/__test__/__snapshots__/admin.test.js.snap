// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render admin 1`] = `
<div
  className="signin"
>
  <div
    className="wide column"
  >
    <h2
      className="ui teal image header"
    >
      <div
        className="content"
      >
        Log-in
      </div>
    </h2>
    <form
      className="ui large form"
      noValidate={undefined}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="user icon"
            />
            <input
              className="field-input"
              disabled={false}
              id="login"
              name="signinForm.login"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Login"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="lock icon"
            />
            <input
              className="field-input"
              disabled={false}
              id="password"
              name="signinForm.password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid large teal submit button"
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        className=""
      >
        <ul
          className="list"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`Render link form & table if authenticated 1`] = `
<div>
  <form
    className="ui form column massive separator"
    noValidate={undefined}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui action input"
      >
        <input
          autoComplete="off"
          className="field-input"
          disabled={false}
          id="link"
          name="linkForm.linkURL"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholder="Past your original url"
          type="text"
          value=""
        />
        <button
          className="ui button"
          type="submit"
        >
          Go !
        </button>
      </div>
    </div>
  </form>
  <div
    className="ReactTable"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
      style={undefined}
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "270px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
          style={undefined}
        >
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Short URL
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Long URL
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -sort-desc -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "25 0 auto",
                "maxWidth": null,
                "width": "25px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Created at
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "15 0 auto",
                "maxWidth": null,
                "width": "15px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Visits
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th  rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "30 0 auto",
                "maxWidth": null,
                "width": "30px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            />
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "270px",
          }
        }
      />
    </div>
    <div
      className="pagination-bottom"
    >
      <div
        className="-pagination"
        style={undefined}
      >
        <div
          className="-previous"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          className="-center"
        >
          <span
            className="-pageInfo"
          >
            Page
             
            <div
              className="-pageJump"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onKeyPress={[Function]}
                type="number"
                value={1}
              />
            </div>
             
            of
             
            <span
              className="-totalPages"
            >
              1
            </span>
          </span>
          <span
            className="select-wrap -pageSizeOptions"
          >
            <select
              onChange={[Function]}
              value={20}
            >
              <option
                value={5}
              >
                5
                 
                rows
              </option>
              <option
                value={10}
              >
                10
                 
                rows
              </option>
              <option
                value={20}
              >
                20
                 
                rows
              </option>
              <option
                value={25}
              >
                25
                 
                rows
              </option>
              <option
                value={50}
              >
                50
                 
                rows
              </option>
              <option
                value={100}
              >
                100
                 
                rows
              </option>
            </select>
          </span>
        </div>
        <div
          className="-next"
        >
          <button
            className="-btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      className="rt-noData"
    >
      No links found
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`Render loading icon if session is ready for auth 1`] = `
<div
  className="loader"
>
  <img
    alt=""
    height="60"
    src="loader.svg"
    width="60"
  />
</div>
`;

exports[`Render loading icon on fetching 1`] = `
<div
  className="loader"
>
  <img
    alt=""
    height="60"
    src="loader.svg"
    width="60"
  />
</div>
`;

exports[`Render signin form is deauthenticated 1`] = `
<div
  className="signin"
>
  <div
    className="wide column"
  >
    <h2
      className="ui teal image header"
    >
      <div
        className="content"
      >
        Log-in
      </div>
    </h2>
    <form
      className="ui large form"
      noValidate={undefined}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="user icon"
            />
            <input
              className="field-input"
              disabled={false}
              id="login"
              name="signinForm.login"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Login"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="lock icon"
            />
            <input
              className="field-input"
              disabled={false}
              id="password"
              name="signinForm.password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid large teal submit button"
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        className=""
      >
        <ul
          className="list"
        />
      </div>
    </form>
  </div>
</div>
`;
